### PR TITLE
Parse embed url as soon as it is pasted

### PR DESCRIPTION
**Pull request by RifRaf44**
When you paste some text into the editor, the event is captured and parsed for youtube or vimeo links. If it contains one of these the embed function is called and the video is embeded into the editor. Both of these functions are tested in the spec file.

One new setting is introduced in the Embeds addon: 
parseOnPaste: false

same as pull request #279 but squashed and cleaned (no dist files)

### CODE REVIEW

**Comment by RifRaf44** [comment]
I implemented the parsePasted using the oEmbed service and checking the URL using [Diego Perini's RegEx ](https://gist.github.com/dperini/729294 ).

File: src/js/embeds.js [hunk 1]
```diff
@@ -294,8 +294,7 @@
         }
         var pastedUrl = e.originalEvent.clipboardData.getData('text');
 
-        if (regexHelper.vimeo_parser(pastedUrl) ||
-            regexHelper.youtube_parser(pastedUrl)) {
+        if (regexHelper.urlParser(pastedUrl)) {
```
Comment (orthes) [inline src/js/embeds.js]: Do we really need to test it this thoroughly? Wouldn't it be enough to test if `pastedUrl` starts with http, https, or // - `/^(http(s?):)?\/\//`? And the rest would be handled by oEmbed or out embed function.

This way we wouldn't have to add new file - `regexHelper.js` and regexp wouldn't look so scary. I just want to keep it simple.

**Comment by orthes** [comment]
Thank you. I have one last comment.

File: src/js/embeds.js (hunk 1 above)
Comment (RifRaf44) [inline src/js/embeds.js]: Ok I can agree to this. I'll remove regexHelper and check for a simple http regular expression.
Sorry it took so long, I was sick for some time and also went on holiday :)

**Comment by orthes** [comment]
@RifRaf44 Cool staff man 👍 . Thanks! I've merged it. I love how it's creating a new paragraph when you paste URL in the middle of existing paragraph. Very convenient.

**Comment by RifRaf44** [comment]
Nice! When are you planning to release a new version? I'd like to be able to get the functionality through bower :)

**Pull request merged**